zkvm/src/types.rs

```diff
@@ -101,7 +102,7 @@ pub enum DataWitness {
     Predicate(Box<PredicateWitness>), // maybe having Predicate and one more indirection would be cleaner - lets see how it plays out
     Commitment(Box<CommitmentWitness>),
     Scalar(Box<Scalar>),
-    Input(Box<(FrozenContract, UTXO)>),
+    Input(Box<(FrozenContract, UTXO, TxID)>),
```

Comment:
Now we have a slight problem: as all variants in an `enum` are necessarily public (unlike struct fields that can be selectively private), someone can construct an inconsistent set of this triplet. To prevent this, it makes sense to have this triplet made into a struct, with private fields `contract, utxo, txid` and a method `.destructure(self) -> (FC, UTXO, TxID)` that consumes self and returns these fields. And all the initializer methods that we might need will ensure the consistency of this thing.

Reply:
(We can get back to what these initializers are in a follow-up PR where we'll attempt to compose a tx)